[RFC][LOGS] Add options to show cutlass logs

henrylhtsang opened the pull request:
Summary:
Use `os.environ["TORCH_LOGS"] = "+cutlass"` to see CUTLASS backend logs.

For example,
```
cutlass-3/python/cutlass_library/manifest.py:731] [0/0] Culled cutlass_tensorop_i168256xorgemm_b1_256x64_1024x4_tn_align128 from manifest
...
cutlass_library/generator.py:58] [0/0] ***   CreateConvOperator3x
cutlass_library/generator.py:58] [0/0] ***     conv_kind: 1
cutlass_library/generator.py:58] [0/0] ***     ConvOperation3x::init: conv_kind: 1
cutlass_library.library.TileDescription object at 0x7feea280c640>
```

Test Plan: tested offline.

Differential Revision: D69082809




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

henrylhtsang commented:
nevermind, os.environ["TORCH_LOGS"] = "+cutlass_library" works already

mlazos commented:
> nevermind, os.environ["TORCH_LOGS"] = "+cutlass_library" works already

I was looking for this, thanks! haha

henrylhtsang commented:
Okay thanks @mlazos for pointing out this doesn't work in OSS.

reopening this PR so `os.environ["TORCH_LOGS"] = "+cutlass"` will work

henrylhtsang commented:
I think this is caused by how cutlass library is handled when we build pytorch from source.

henrylhtsang commented:
so now use os.environ["TORCH_LOGS"] = "+cutlass" for cutlass related logs

mlazos commented:
Looks good, can you also add a test in test/dynamo/test_logging.py

henrylhtsang committed:
[RFC][LOGS] Add options to show cutlass logs (#146365)

Summary:

Use `os.environ["TORCH_LOGS"] = "+cutlass"` to see CUTLASS backend logs.

For example,
```
cutlass-3/python/cutlass_library/manifest.py:731] [0/0] Culled cutlass_tensorop_i168256xorgemm_b1_256x64_1024x4_tn_align128 from manifest
...
cutlass_library/generator.py:58] [0/0] ***   CreateConvOperator3x
cutlass_library/generator.py:58] [0/0] ***     conv_kind: 1
cutlass_library/generator.py:58] [0/0] ***     ConvOperation3x::init: conv_kind: 1
cutlass_library.library.TileDescription object at 0x7feea280c640>
```

Test Plan: tested offline.

Differential Revision: D69082809

henrylhtsang commented:
abandoning after talking with @mlazos , since not sure if enough people care about logs from cutlass.
